drivers/usb/gadget/function/f_scm.c

```diff
@@ -0,0 +1,574 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * f_scm.c -- USB Socket Control Model (SCM) function driver
+ *
+ * Copyright (C) 2018-2019 Xaptum, Inc.
+ */
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/device.h>
+#include <linux/usb/composite.h>
+#include "u_scm.h"
+
+/*
+ * The Socket Control Model is protocol for a USB device to manage and use
+ * Berkeley/POSIX-style sockets on the host.  It allows the device to
+ * communicate with remote servers using the network connection of the host.
+ * Currently supported protocols include TCP/IPv4 and TCP/IPv6.
+ *
+ * The SCM data transfer model uses bulk endpoints (short packet terminated)
+ * to send and recieve data from the hosts sockets and interrupt endpoints to
+ * communcate socket management data (open, connect, ect).
+ */
+
+#define MAX_INT_PACKET_SIZE    64
+#define SCM_SUBCLASS 0xab
+#define SCM_STATUS_INTERVAL_MS 32
+
+/**
+ * Usb function structure definition
+ */
+struct f_scm {
+	struct usb_function function;
+
+	struct usb_ep *bulk_in;
+	struct usb_ep *bulk_out;
+	struct usb_ep *cmd_out;
+	struct usb_ep *cmd_in;
+	struct usb_ep *ep0;
+};
+
+/*
+ * The USB interface descriptor to tell the host
+ * how many endpoints are being deviced, ect.
+ */
+static struct usb_interface_descriptor scm_intf = {
+	.bLength            = sizeof(scm_intf),
+	.bDescriptorType    = USB_DT_INTERFACE,
+
+	.bNumEndpoints      = 4,
+	.bInterfaceClass    = USB_CLASS_VENDOR_SPEC,
+	.bInterfaceSubClass = SCM_SUBCLASS,
```

Comment:
I like that the other gadget functions add comments for the dynamic fields that are in the descriptor --- those that are set in the `bind` function.  For example, [here](https://elixir.bootlin.com/linux/v4.4/source/drivers/usb/gadget/function/f_ecm.c#L127).  That make its obvious which fields will be set in the actual descriptor that is sent.

Could you please add the same here.  I think it is just
- `/* .bInterfaceNumber = DYNAMIC */` in `scm_intf`.
- `/* .iInterface = DYNAMIC */` in `scm_intf`.